vectorlink-worker/src/main.rs

```diff
@@ -29,7 +64,11 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         args.identity.unwrap_or_else(generate_identity),
     )
     .await?;
+    successful_connection_counter.inc();
+
     VectorlinkTaskHandler::process_queue(&mut queue).await?;
```

Comment:
I actually think it's probably far more useful to do all (most?) prometheus processing inside of this function. It could even just spawn the endpoint, cause we want all queue processors to just have this.

In there, there will be a loop that actually does the task handling, and that is where keeping counters would be most important.

Reply:
(the function being `TaskHandler::process_queue`)

Reply:
@matko I'm confused by this import.

- The import at the top is for `handler::VectorlinkTaskHandler`
- `vectorlink-worker/src/handler.rs` defines VectorlinkTaskHandler and implements TaskHandler for it, but only contains the `process` function, not `process_queue`
- `vectorlink-task/src/task.rs` defines the TaskHandler trait (presumably the one that is implemented on VectorlinkTaskHandler?), which does contain `process_queue`, but I don't see how it's being imported and called through `VectorlinkTaskHandler`

I feel like the logging should be put into `vectorlink-task/src/task.rs > process_queue`, as you stated in your comment, but can you help me understand how it gets from there to the main() function in vectorlink-worker?